src/Service/RegistrationService.php

```diff
@@ -2,49 +2,130 @@
 namespace App\Service;
 
 use App\Entity\User;
+use App\Entity\VerifyUser;
 use App\Exception\ProfilePictureUploadException;
 use Doctrine\ORM\EntityManagerInterface;
 use Symfony\Component\HttpFoundation\File\UploadedFile;
 use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
 use Symfony\Component\PasswordHasher\Hasher\UserPasswordHasherInterface;
 use Symfony\Component\HttpFoundation\File\Exception\FileException;
 use Psr\Log\LoggerInterface;
+use Symfony\Component\Mailer\MailerInterface;
+use Symfony\Component\Mime\Email;
+use Symfony\Bridge\Twig\Mime\TemplatedEmail;
+use Symfony\Component\Mime\Address;
+use Symfony\Component\HttpFoundation\Response;
 
 class RegistrationService
 {
-
     public function __construct(
         private readonly EntityManagerInterface $entityManager, 
         private readonly UserPasswordHasherInterface $passwordHasher, 
         private readonly ParameterBagInterface $params,
-        private readonly LoggerInterface $logger 
-    )
-    {}
+        private readonly LoggerInterface $logger,
+        private readonly MailerInterface $mailer,
+        private readonly string $appHost,
+    ) {}
 
-    public function registerUser(User $user, string $plainPassword, ?UploadedFile $profilePicture): void
+    public function registerUser(User $user, string $plainPassword, $profilePicture): Response
     {
         // Set the user password
         $user->setPassword($this->passwordHasher->hashPassword($user, $plainPassword));
-        
-        // Handle the profile picture upload
         if ($profilePicture) {
-            $this->uploadProfilePicture($user, $profilePicture);
+            try {
+                $this->uploadProfilePicture($user, $profilePicture);
+            } catch (ProfilePictureUploadException $e) {
+                $this->logger->error($e->getAdditionalErrorInfo());
+                return new Response($e->getMessage(), 400);
+            }
         }
+        // Set user as unverified initially
+        $user->setVerified(false);
+        $user->setRoles(['ROLE_USER']); // Initially ROLE_USER
+        
+        // Generate verification token
+        $token = bin2hex(random_bytes(16));
+
+        // Create and persist VerifyUser entity
+        $verifyUser = new VerifyUser();
+        $verifyUser->setUser($user);
+        $verifyUser->setToken($token);
+        $this->entityManager->persist($verifyUser);
 
-        // Save the user to the database
+        // Persist User entity
         try {
             $this->entityManager->persist($user);
             $this->entityManager->flush();
 
+            // Send verification email
+            $this->sendVerificationEmail($user, $token);
+
             // Log the successful registration
             $this->logger->info('User registered successfully', ['email' => $user->getEmail()]);
+
+            return new Response('Registration successful! Please check your email for verification.', 200);
```

Comment:
don't return response in the service, return $message or throw an exception and only return response in controllers.